assertions/src/main/java/org/jboss/arquillian/graphene/assertions/WebElementAssert.java

```diff
@@ -1,18 +1,208 @@
 package org.jboss.arquillian.graphene.assertions;
 
 import org.assertj.core.api.AbstractAssert;
+import org.jboss.arquillian.graphene.findby.ByJQuery;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class WebElementAssert extends AbstractAssert<WebElementAssert, WebElement> {
 
+    private boolean negated;
+
     public WebElementAssert(WebElement actual) {
         super(actual, WebElementAssert.class);
     }
 
     public WebElementAssert hasText(String expectedText) {
-        assertThat(this.actual.getText()).isEqualTo(expectedText);
+        if(this.negated){
+            assertThat(this.actual.getText()).as("checking strings are not equal").overridingErrorMessage("The content of the element should not be equal to " + expectedText +", but it is.").isNotEqualTo(expectedText);
+            this.negated = false;
+            return this;
+        }
+        else {
+            assertThat(this.actual.getText()).as("checking equality").overridingErrorMessage("The content of the element should be equal to" + expectedText + "but it is not.").isEqualTo(expectedText);
+            return this;
+        }
+    }
+
+    public WebElementAssert hasChild(){
+            if (this.negated) {
+                assertThat(this.actual.findElements(ByJQuery.selector("*"))).asList().size().isEqualTo(0);
+                this.negated = false;
```

Comment:
If the about fails this line would never be reached. Why are we doing this assignment in the first place?